### PR TITLE
build-cache: don't hardcode the distro used

### PR DESCRIPTION
**Pull request by simondeziel**

### CODE REVIEW

File: bin/build-cache [hunk 1]
```diff
@@ -34,14 +34,15 @@ ARCHIVE="http://us.ports.ubuntu.com/ubuntu-ports"
 if [ "${ARCH}" = "amd64" ] || [ "${ARCH}" = "i386" ]; then
     ARCHIVE="http://us.archive.ubuntu.com/ubuntu"
 fi
+DISTRO="$(lsb_release -sc)"
```
[the remark this answers is not in the document]
Comment (stgraber) [reply bin/build-cache:37]: The `${ARCH}` right before is actually proof that it's needed as ARCH isn't defined anywhere in the script being run in the container, instead it gets that value from the parent context.

**Comment by tomponline** [comment]
Your commit name isnt right I think

File: bin/build-cache (hunk 1 above)
[the remark this answers is not in the document]
Comment (stgraber) [reply bin/build-cache:37]:
```
stgraber@castiana:~$ ./test.sh 
Creating f1
Starting f1                                   
kinetic                        
+ DISTRO=kinetic
+ echo kinetic
stgraber@castiana:~$ cat test.sh
lxc launch images:ubuntu/focal f1
(
    cat << EOF
#!/bin/sh
set -eux
DISTRO="$(lsb_release -sc)"
echo \${DISTRO}
EOF
) | lxc exec f1 -- bash
lxc delete -f f1
stgraber@castiana:~$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 22.10
Release:	22.10
Codename:	kinetic
stgraber@castiana:~$ 
```

[the remark this answers is not in the document]
Comment (simondeziel) [reply bin/build-cache]: And I stand corrected, thanks!

**Pull request merged**